Solution1/Solution1.Tests/BL/Services/WeatherApiServiceTests.cs

```diff
@@ -1,60 +1,125 @@
         public async Task GetByCityNameAsync_ReturnedWeatherApiDTO_Success()
         {
             // Arrange
-            var cityName = "Minsk";
             var urlString = "https://api.openweathermap.org/data/2.5/weather?q=Minsk&appid=3fe39edadae3ae57d133a80598d5b120&units=metric";
-
+            
             var response = new HttpResponseMessage
             {
                 StatusCode = HttpStatusCode.OK,
-                Content = new StringContent(@"{ ""main"": {""temp"": 1.86}, ""name"": ""Minsk""}"),
+                Content = new StringContent(JsonSerializer.Serialize(new { Main = new { Temp = temp }, Name = cityName }, _serializerOptions)),
             };
 
-            _httpMessageHandler
-               .Protected()
-               .Setup<Task<HttpResponseMessage>>(
-                  "SendAsync",
-                  ItExpr.Is<HttpRequestMessage>(
-                      request => 
-                      request.Method == HttpMethod.Get
-                      && request.RequestUri.ToString() == urlString),
-                  ItExpr.IsAny<CancellationToken>())
-               .ReturnsAsync(response);
-
+            _httpMessageHandler.GetSettings(response, urlString);
+            
             // Act
             var result = await _weatherApiService.GetByCityNameAsync(cityName);
 
             // Assert
-            var expectedWeatherApiDto = new WeatherApiDTO() { CityName = cityName, TemperatureValues = new WeatherApiTempDTO() { Temp = 1.86 } };
+            var expectedWeatherApiDto = new WeatherApiDTO() { CityName = cityName, TemperatureValues = new WeatherApiTempDTO() { Temp = temp } };
             Assert.True(new CompareLogic().Compare(expectedWeatherApiDto, result).AreEqual);
         }
+
+        [Fact]
+        public async Task GetForecastByCityNameAsync_ReturnedForecastWeatherApiDTO_Success()
+        {
+            // Arrange
+            var lat = 53;
+            var lon = 27;
+            var dataTime = new DateTime(2022, 03, 05, 9, 0, 0);
+            var countWeatherPoints = 2;
+            var pointsPeriod = 3;
+
+            var urlCoordinates = $"http://api.openweathermap.org/geo/1.0/direct?q={cityName}&appid=3fe39edadae3ae57d133a80598d5b120";
+            var urlForecast = $"https://api.openweathermap.org/data/2.5/forecast?lat={lat}&lon={lon}&cnt=2&units=metric&appid=3fe39edadae3ae57d133a80598d5b120";
+
+            var responseCoordinates = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = new StringContent(JsonSerializer.Serialize(new[] { new { Name = cityName, Lat = lat, Lon = lon } }, _serializerOptions)),
+            };
+
+            var listWeatherAnonymousObject = Enumerable.Range(0, 0)
+                .Select(a => new { DateTime = default(string), Main = new { Temp = default(double) } }).ToList();
+
+            for (int currentCountTempPoints = 0; currentCountTempPoints < countWeatherPoints; currentCountTempPoints++)
```

Comment:
you don't need any logic here. The method which you test just should return ANY data, it can be aaaaaa data, it's NOT an integration test, it can be completely random data